TF_Service_dotNet/TouchFree.Tooling/InputActionManager.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Linq;
+using Ultraleap.TouchFree.Library;
+
+namespace Ultraleap.TouchFree.Tooling
+{
+    // Class: InputActionManager
+    // The manager for all <InputActions> to be handled and distributed. This runs the
+    // received data through any referenced <InputActionPlugins> and finaly distributes the data
+    // via the  <TransmitInputAction> event which should be listened to by any class hoping to make
+    // use of incoming <InputActions>.
+    public class InputActionManager
+    {
+        // Delegate: InputActionEvent
+        // An Action to distribute a <InputAction> via the <TransmitInputAction> event listener.
+        public delegate void InputActionEvent(ClientInputAction _inputData);
+
+        // Variable: TransmitInputAction
+        // An event for transmitting <InputActions> that have been modified via the active
+        // <plugins>
+        public static event InputActionEvent TransmitInputAction;
+
+        // Variable: TransmitRawInputAction
+        // An event for transmitting <InputActions> that have NOT been modified via any
+        // <plugins>
+        public static event InputActionEvent TransmitRawInputAction;
+
+        public static InputActionManager Instance;
+
+        // Variable: plugins
+        // A pre-defined plugin array of <ToggleablePlugins> that modify incoming <InputActions>
+        // based on custom rules.
+        ToggleablePlugin[] plugins;
+
+        public void Awake()
```

Comment:
The awake method is a hangover from Unity, and seems to make for an awkward workflow when using this lib directly. Could this be exchanged for a constructor?